library/sql.inc

```diff
@@ -123,18 +123,121 @@ class ADODB_mysqli_log extends ADODB_mysqli
+                    "${GLOBALS['OE_SITE_DIR']}/documents/certificates/mysql-ca",
+                    null,
+                    null
+                );
+            } else {
+                // without client side certificate/key
+                error_log("${GLOBALS['OE_SITE_DIR']}/documents/certificates/mysql-ca");
+                mysqli_ssl_set(
+                    $this->_connectionID,
+                    null,
+                    null,
+                    "${GLOBALS['OE_SITE_DIR']}/documents/certificates/mysql-ca",
+                    null,
+                    null
+                );
+            }
+        }
+
+        #if (!empty($this->port)) $argHostname .= ":".$this->port;
+        $ok = mysqli_real_connect(
+            $this->_connectionID,
+            $argHostname,
+            $argUsername,
+            $argPassword,
+            $argDatabasename,
+            # PHP7 compat: port must be int. Use default port if cast yields zero
+            (int)$this->port != 0 ? (int)$this->port : 3306,
+            $this->socket,
+            $this->clientFlags
+        );
+
+        if ($ok) {
+            if ($argDatabasename) {
+                return $this->SelectDB($argDatabasename);
+            }
+            return true;
+        } else {
+            if ($this->debug) {
+                ADOConnection::outp("Could't connect : "  . $this->ErrorMsg());
+            }
+            $this->_connectionID = null;
+            return false;
+        }
+    }
 }
 if (!defined('ADODB_FETCH_ASSOC')) {
     define('ADODB_FETCH_ASSOC', 2);
 }
 
 $database = NewADOConnection("mysqli_log"); // Use the subclassed driver which logs execute events
-// Below clientFlags flag is telling the mysql connection to allow local_infile setting,
+// Below optionFlags flag is telling the mysql connection to ensure local_infile setting,
 // which is needed to import data in the Administration->Other->External Data Loads feature.
-// Note this is a specific bug to work in Ubuntu 12.04, of which the Data Load feature does not
-// work and is suspicious for a bug in PHP of that OS; Setting this clientFlags fixes this bug
-// and appears to not cause problems in other operating systems.
-$database->clientFlags = 128;
+// (Note the MYSQLI_READ_DEFAULT_GROUP is just to keep the current setting hard-coded in adodb)
+$database->optionFlags = array(array(MYSQLI_READ_DEFAULT_GROUP,0), array(MYSQLI_OPT_LOCAL_INFILE,1));
+// Set mysql to use ssl, if applicable.
+// Can support basic encryption by including just the mysql-ca pem (this is mandatory for ssl)
+// Can also support client based certificate if also include mysql-cert and mysql-key (this is optional for ssl)
+if (file_exists($GLOBALS['OE_SITE_DIR'] . "/documents/certificates/mysql-ca")) {
+    $database->clientFlags = MYSQLI_CLIENT_SSL;
+    error_log("DEBUG: Passing MYSQLI_CLIENT_SSL flag");
+}
 $database->port = $port;
 $database->PConnect($host, $login, $pass, $dbase);
```

Comment:
Will likely need to change this to Connect (ie. not a persistent connection) to support ssl